pkg/server/skydb/pq/builder/factory.go

```diff
@@ -229,7 +230,7 @@ func (f *predicateSqlizerFactory) newExpressionSqlizer(expr skydb.Expression) (e
 		`unexpected expression type`)
 }
 
-func (f *predicateSqlizerFactory) newExpressionSqlizerForKeyPath(expr skydb.Expression) (expressionSqlizer, error) {
+func (f *predicateSqlizerFactory) newExpressionSqlizerForKeyPath(expr skydb.Expression, addJoinedFieldToExtraColumn bool) (expressionSqlizer, error) {
```

Comment:
See `Avoid Side Effects` in https://medium.com/javascript-scene/the-single-biggest-mistake-programmers-make-every-day-62366b432308.

In short, if a function returns something except error, it is expected not to have any side effects.

Reply:
Create another method for sorting instead.